src/manifest.rs

```diff
@@ -0,0 +1,30 @@
+use anyhow::anyhow;
+use std::path::{Path, PathBuf};
+
+use crate::opts::DEFAULT_MANIFEST_FILE;
+
+pub(crate) fn manifest_path(path: impl AsRef<Path>) -> anyhow::Result<PathBuf> {
+    let path = path.as_ref();
+
+    if path.is_file() {
+        Ok(path.to_owned())
+    } else if path.is_dir() {
+        let file_path = path.join(DEFAULT_MANIFEST_FILE);
+        if file_path.is_file() {
+            Ok(file_path)
+        } else {
+            Err(anyhow!(
+                "Directory {} does not contain a file named 'spin.toml'",
+                path.display()
+            ))
+        }
+    } else {
+        let pd = path.display();
+        let err = match path.try_exists() {
+            Err(e) => anyhow!("Error accessing path {pd}: {e:#}"),
+            Ok(false) => anyhow!("No such file or directory '{pd}'"),
+            Ok(true) => anyhow!("Path {pd} is neither a file nor a directory"),
```

Comment:
Seems like this should only happen on broken symbolic links. Maybe referencing that a link is broken might be more clear?

Reply:
Or missing permissions on a parent dir, or (presumably) disk/fs error.

Reply:
This error message, which already existed, seems in danger of turning into a copy of _Troubleshooting the Unix (R) Operating System_ (Addison Wesley, 1991)...!  I'm happy to adopt concrete phrasing suggestions if we can nut one out quickly; otherwise I suggest we leave it as is and tackle improving it as a separate piece of work.